newIDE/app/src/ObjectGroupEditor/EditedObjectGroupEditorDialog.js

```diff
@@ -0,0 +1,211 @@
+import useForceUpdate from '../Utils/UseForceUpdate';
+import { Tabs } from '../UI/Tabs';
+import { Column, Line } from '../UI/Grid';
+import { ProjectScopedContainersAccessor } from '../InstructionOrExpression/EventsScope.flow';
+import useDismissableTutorialMessage from '../Hints/useDismissableTutorialMessage';
+import VariablesList from '../VariablesList/VariablesList';
+
+const gd: libGDevelop = global.gd;
+
+export type ObjectGroupEditorTab = 'objects' | 'variables';
+
+type Props = {|
+  project: gdProject,
+  projectScopedContainersAccessor: ProjectScopedContainersAccessor,
+  group: gdObjectGroup,
+  onApply: () => void,
+  onCancel: () => void,
+  globalObjectsContainer: gdObjectsContainer | null,
+  objectsContainer: gdObjectsContainer,
+  initialTab: ?ObjectGroupEditorTab,
+|};
+
+const EditedObjectGroupEditorDialog = ({
+  project,
+  projectScopedContainersAccessor,
+  group,
+  onApply,
+  onCancel,
+  globalObjectsContainer,
+  objectsContainer,
+  initialTab,
+}: Props) => {
+  const forceUpdate = useForceUpdate();
+  const {
+    onCancelChanges,
+    notifyOfChange,
+  } = useSerializableObjectCancelableEditor({
+    serializableObject: group,
+    onCancel,
+  });
+
+  const [currentTab, setCurrentTab] = React.useState<ObjectGroupEditorTab>(
+    initialTab || 'objects'
+  );
+
+  const groupVariablesContainer = ((React.useRef<gdVariablesContainer | null>(
+    null
+  ): any): { current: gdVariablesContainer });
+  if (!groupVariablesContainer.current) {
+    // The VariablesContainer is returned by value.
+    // Thus, the same instance is reused every time.
+    groupVariablesContainer.current = gd.GroupVariableHelper.mergeVariableContainers(
+      projectScopedContainersAccessor.get().getObjectsContainersList(),
+      group
+    );
+  }
+
+  const {
+    notifyOfChange: notifyOfVariableChange,
+    getOriginalContentSerializedElement: getOriginalVariablesSerializedElement,
+  } = useSerializableObjectCancelableEditor({
+    serializableObject: groupVariablesContainer.current,
+    onCancel: () => {},
+    resetThenClearPersistentUuid: true,
+  });
+
+  const apply = async () => {
+    onApply();
+
+    const originalSerializedVariables = getOriginalVariablesSerializedElement();
+    const changeset = gd.WholeProjectRefactorer.computeChangesetForVariablesContainer(
+      originalSerializedVariables,
+      groupVariablesContainer.current
+    );
+
+    gd.WholeProjectRefactorer.applyRefactoringForGroupVariablesContainer(
+      project,
+      globalObjectsContainer || objectsContainer,
+      objectsContainer,
+      groupVariablesContainer.current,
+      group,
+      changeset,
+      originalSerializedVariables
+    );
+    groupVariablesContainer.current.clearPersistentUuid();
+  };
+
+  const [groupObjectNames, setGroupObjectNames] = React.useState<Array<string>>(
+    () => group.getAllObjectsNames().toJSArray()
+  );
```

Comment:
Is this worth duplicating this information in the state? Any duplication runs the risk of being desynchronized if someones forget to call setGroupObjectNames.

It might be slightly less efficient, but more bug-prone, to just use `group.getAllObjectsNames().toJSArray()` when you need it.

If it's an optimisation, then rename it so that it's obvious for the reader `cachedGroupObjectNames` and add a comment explaining why "// Cache the object names to avoid recomputing them at every render".